### PR TITLE
Map TestSetGetRoundTrip

### PR DESCRIPTION
**Pull request by gdbelvin**
#1740

- [ ] I have updated the [CHANGELOG](CHANGELOG.md).
  - Adjust the draft version number according to [semantic versioning](https://semver.org/) rules.
- [ ] I have updated [documentation](docs/) accordingly (including the [feature implementation matrix](docs/Feature_Implementation_Matrix.md)).

### CODE REVIEW

**Comment by gdbelvin** [comment]
Famous last words -- but the last PR of this batch?

File: integration/storagetest/maptests.go [hunk 1]
```diff
@@ -193,3 +196,64 @@ func (*mapTests) TestTileRoundTrip(ctx context.Context, t *testing.T, ms storage
 		t.Fatalf("ReadWriteTransaction(): %v", err)
 	}
 }
+
+func (*mapTests) TestGetSetRoundTrip(ctx context.Context, t *testing.T, ms storage.MapStorage, as storage.AdminStorage) {
```
Comment (mhutchinson) [inline integration/storagetest/maptests.go:200]:
```suggestion
func (*mapTests) TestSetGetRoundTrip(ctx context.Context, t *testing.T, ms storage.MapStorage, as storage.AdminStorage) {
```

You need to set before you can get :-)

File: integration/storagetest/maptests.go [hunk 2]
```diff
@@ -193,3 +196,64 @@ func (*mapTests) TestTileRoundTrip(ctx context.Context, t *testing.T, ms storage
...
+	tree := mustCreateTree(ctx, t, as, storageto.MapTree)
+	mustSignAndStoreMapRoot(ctx, t, ms, tree, &types.MapRootV1{Revision: uint64(0)})
+
+	num := int64(10)
```
Comment (mhutchinson) [inline integration/storagetest/maptests.go:204]: Just inline this? I'm not sure the documentation provided by `num` materially decreases the magic number factor here.

File: integration/storagetest/maptests.go [hunk 3]
```diff
@@ -193,3 +196,64 @@ func (*mapTests) TestTileRoundTrip(ctx context.Context, t *testing.T, ms storage
...
+		}
+		if got, want := len(leaves), 1; got != want {
+			t.Errorf("Get(rev: %d): %d leaves, want %d", rev, got, want)
+			continue
+		}
+		if got := leaves[0]; !cmp.Equal(got, want, protocmp.Transform()) {
+			t.Errorf("Get(rev: %d) = diff:\n%v", rev, cmp.Diff(got, want, protocmp.Transform()))
+			continue
+		}
+		if err := tx.Commit(ctx); err != nil {
```
Comment (mhutchinson) [inline integration/storagetest/maptests.go:255]: The commit here was surprising in a read-only test, but worth testing it doesn't blow  up I guess.

File: integration/storagetest/maptests.go (hunk 1 above)
Comment (gdbelvin) [reply integration/storagetest/maptests.go:200]: :-)

File: integration/storagetest/maptests.go (hunk 2 above)
Comment (gdbelvin) [reply integration/storagetest/maptests.go:204]: done

File: integration/storagetest/maptests.go (hunk 3 above)
Comment (gdbelvin) [reply integration/storagetest/maptests.go:255]: Hmm, good point. At the very least it should be outside the loop

**Pull request merged**